pallets/gear-built-in-actors/src/lib.rs

```diff
@@ -0,0 +1,443 @@
+use gear_core::{
+    ids::{MessageId, ProgramId},
+    message::{ReplyMessage, ReplyPacket, StoredDispatch},
+};
+use gear_core_errors::SimpleExecutionError;
+use pallet_gear::BuiltInActor;
+use pallet_staking::RewardDestination;
+use parity_scale_codec::{Decode, Encode};
+use scale_info::TypeInfo;
+use sp_io::hashing::blake2_256;
+use sp_runtime::traits::{StaticLookup, TrailingZeroInput, UniqueSaturatedInto};
+use sp_std::{collections::btree_set::BTreeSet, prelude::*};
+
+pub use pallet::*;
+
+pub type BalanceOf<T> = <T as pallet_staking::Config>::CurrencyBalance;
+
+const LOG_TARGET: &str = "gear::built-in-actor";
+
+// TODO: update upon benchmarking
+const BASE_GAS_BURNED: u64 = 0;
+
+/// Available actor types.
+#[derive(Decode, Encode, Debug, Clone, Copy, PartialEq, Eq, Sequence, TypeInfo)]
+#[repr(u32)]
+pub enum ActorType {
+    StakingProxy = 100,
+}
+
+/// Built-in actor error
+#[derive(Encode, Decode, Debug, PartialEq, Eq, PartialOrd, Ord, derive_more::Display)]
+pub enum BuiltInActorReason {
+    /// Not enough gas to dispatch a call.
+    #[display(fmt = "Not enough gas to dispatch a call")]
+    InsufficientGas,
+    /// Error transferring value due to insufficient funds or existence requirements violation.
+    #[display(fmt = "Error transferring value")]
+    TransferError,
+    /// Protocol error (encoding/decoding)
+    #[display(fmt = "Failure to decode message")]
+    UnknownMessageType,
+}
+
+impl From<BuiltInActorReason> for SimpleExecutionError {
+    /// Convert [`BuiltInActorReason`] into [`gear_core_errors::SimpleExecutionError`].
+    // TODO: think of a better mapping.
+    fn from(reason: BuiltInActorReason) -> Self {
+        match reason {
+            BuiltInActorReason::InsufficientGas => SimpleExecutionError::RanOutOfGas,
+            BuiltInActorReason::TransferError => SimpleExecutionError::UserspacePanic,
+            BuiltInActorReason::UnknownMessageType => SimpleExecutionError::UserspacePanic,
+        }
+    }
+}
+
+#[frame_support::pallet]
+pub mod pallet {
+    use super::*;
+    use frame_support::pallet_prelude::*;
+    use frame_system::pallet_prelude::*;
+
+    /// The current storage version.
+    const STORAGE_VERSION: StorageVersion = StorageVersion::new(0);
+
+    #[pallet::pallet]
+    #[pallet::storage_version(STORAGE_VERSION)]
+    #[pallet::without_storage_info]
+    pub struct Pallet<T>(PhantomData<T>);
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config + pallet_staking::Config {
+        /// The overarching event type.
+        type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+
+        /// The overarching call type.
+        type RuntimeCall: Parameter
+            + Dispatchable<RuntimeOrigin = Self::RuntimeOrigin, PostInfo = PostDispatchInfo>
+            + GetDispatchInfo
+            + From<pallet_staking::Call<Self>>;
+
+        /// The staking proxy actor pallet id, used for deriving its sovereign account ID.
```

Comment:
Perhaps comment should be edited since the id is used to derive not only staking actor ID but for all future actors too